go/vt/discovery/tablet_health_check.go

```diff
@@ -177,6 +177,12 @@ func (thc *tabletHealthCheck) processResponse(hc *HealthCheckImpl, shr *query.St
 		serving = false
 	}
 
+	ti, err := hc.ts.GetTablet(thc.ctx, thc.Tablet.Alias)
+	if err != nil {
+		return fmt.Errorf("could not get tablet info from topo: %v", err)
+	}
+	thc.Tablet = ti.Tablet
```

Comment:
This might lead to a high impact on performance since we have to question the topo for every health update. However, we can remove the commit that introduces this, the bug is fixable with only one or two of the two fixes proposed by this PR.

Reply:
We should not do topo calls here as you rightly said it will have a performance impact.
Also, it will increase load on topo server. 
If we plan to fix this, we should do it in a separate PR and also understand how to do it better.

Reply:
@frouioui @harshit-gangal Wondering about this, if the specific state of a tablet here is not regularly updated, should that state be removed also from the object? 

This is more for a future more deeper change, but I wonder if it's better to not make the information available at all if it can't be relied on. That way it won't be used in other places and no new usage would be added that might be unknowingly broken? 

Not sure how hard this would be to change though but avoiding making an easy mistake might be worth it?

Reply:
I think you're right @dbussink. I feel like both variables serve two different purposes, it is worth evaluating whether this can be removed or not.

Reply:
This is complex. @harshit-gangal and I looked at the code to see whether we can completely remove the reference to the Tablet struct. We are using other information from the Tablet struct in various places, so we can't do that easily.

Reply:
@deepthi Do you think it would work to create two separate types for this? One with only the static information that can still be depended on in places where we do today and where we don't need access to information that changes (like the tablet replica / primary state).

And then have a separate type for the information that does change and gets updated by health checks?